sdk/eventhub/azure-eventhubs/azure/eventhub/_eventprocessor/sqlite3_partition_manager.py

```diff
@@ -0,0 +1,164 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# -----------------------------------------------------------------------------------
+
+import time
+import uuid
+import sqlite3
+import logging
+from .partition_manager import PartitionManager
+
+logger = logging.getLogger(__name__)
+
+
+def _check_table_name(table_name: str):
+    for c in table_name:
+        if not (c.isalnum() or c == "_"):
+            raise ValueError("Table name \"{}\" is not in correct format".format(table_name))
+    return table_name
+
+
+class Sqlite3PartitionManager(PartitionManager):
+    """An implementation of PartitionManager by using the sqlite3 in Python standard library.
+    Sqlite3 is a mini sql database that runs in memory or files.
+    Please don't use this PartitionManager for production use.
+
+
+    """
+    primary_keys_dict = {"fully_qualified_namespace": "text", "eventhub_name": "text",
+                         "consumer_group_name": "text", "partition_id": "text"}
+    primary_keys = list(primary_keys_dict.keys())
+
+    ownership_data_fields_dict = {"owner_id": "text", "last_modified_time": "real", "etag": "text"}
+    ownership_fields_dict = {**primary_keys_dict, **ownership_data_fields_dict}
+    ownership_data_fields = list(ownership_data_fields_dict.keys())
+    ownership_fields = primary_keys + ownership_data_fields
+
+    checkpoint_data_fields_dict = {"sequence_number": "integer", "offset": "text"}
+    checkpoint_data_fields = list(checkpoint_data_fields_dict.keys())
+    checkpoint_fields_dict = {**primary_keys_dict, **checkpoint_data_fields_dict}
+    checkpoint_fields = primary_keys + checkpoint_data_fields
+
+    def __init__(self, db_filename: str = ":memory:",
+                 ownership_table: str = "ownership", checkpoint_table: str = "checkpoint"):
+        """
+
+        :param db_filename: name of file that saves the sql data.
+         Sqlite3 will run in memory without a file when db_filename is ":memory:".
+        :param ownership_table: The table name of the sqlite3 database.
+        """
+        super(Sqlite3PartitionManager, self).__init__()
+        self.ownership_table = _check_table_name(ownership_table)
+        self.checkpoint_table = _check_table_name(checkpoint_table)
+        conn = sqlite3.connect(db_filename, check_same_thread=False)
+        c = conn.cursor()
+        try:
+            ownership_sql = "create table if not exists " + self.ownership_table\
```

Comment:
never build a SQL query by concatenation, use parameter substitution itself.

Reply:
Putting aside the decision to remove this part for this discussion for now. 
This is a good point. I tried to parameterize this sql but failed. I think the reason is because it's a DDL (like create table) instead of a DML (like insert, update, delete, select).
My understanding is DDL usually can't be parameterized. To guard against sql injection, the code checks input argument ownership_table for any characters that should not be used as a table name.